### PR TITLE
Load 'node-pty' async (fix #105345)

### LINKED ISSUE #105345: Load 'node-pty' async
I see no reason why `node-pty` needs to be loaded sync, as I think it is only used when an integrated terminal is opened. Loading it via `await import('node-pty')` could save some ms on startup:

image

//cc @jrieken

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #105345

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/node/terminalProcess.ts [hunk 1]
```diff
@@ -136,10 +136,10 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 		return undefined;
 	}
 
-	private setupPtyProcess(shellLaunchConfig: IShellLaunchConfig, options: pty.IPtyForkOptions): void {
+	private async setupPtyProcess(shellLaunchConfig: IShellLaunchConfig, options: pty.IPtyForkOptions): Promise<void> {
 		const args = shellLaunchConfig.args || [];
 		this._logService.trace('IPty#spawn', shellLaunchConfig.executable, args, options);
-		const ptyProcess = pty.spawn(shellLaunchConfig.executable!, args, options);
+		const ptyProcess = (await import('node-pty')).spawn(shellLaunchConfig.executable!, args, options);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/node/terminalProcess.ts:142]: Looks good, you can also change the import to `import type` to make sure we don't accidentally import the code again?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/node/terminalProcess.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/node/terminalProcess.ts:142]: nvm, I did it 🙂.

**Review by Tyriar — APPROVED** [review]
Good to merge after CI compile

**Comment by bpasero** [comment]
@Tyriar oh nice, learnt something (`import type...`)

**Pull request merged**